cppcheck seems to work but no lint results appear

I get the same question described [here](https://stackoverflow.com/questions/50984107/cppcheck-seems-to-work-but-no-lint-results-appear-subilme-text-3-mac).
In short, in sublime console, cppcheck works just fine. However, when running `SublimeLinter: Show all errors` command in `ctrl + shift + p`, it shows `No lint results.` 

I'm using win7 & sublime text  3 build 3143 & SublimeLinter v4.6.6 & SublimeLinter-cppcheck v1.1.1.

eh... since the source code for sublimelinter-cppcheck is quite short, i took sometime reading it and found the bug..
my test code 
```cpp
#include <stdio.h>

int main(){
	1111
}
```

the `errormsg` generated by cppcheck in my test code is:
`D:\cplusplus\helloworld\example.cc:4:2: warning: Redundant code: Found a statement that begins with numeric constant. [constStatement]`

but the regex in SublimeLinter-cppcheck is 
```
regex = (
        r'^(?P<file>.+):(?P<line>\d+):\s+'
        r'((?P<error>error)|(?P<warning>warning|style|performance|portability|information)):\s+'
        r'(?P<message>.+)'
    )
```
which means, we cannot get the correct `line` from `re.compile(regex).match(errormsg)`……
change regex to 
```python
regex = (
    r'^(?P<file>.+):(?P<line>\d+):(\d+):\s+'
    r'((?P<error>error)|(?P<warning>warning|style|performance|portability|information)):\s+'
    r'(?P<message>.+)'
)
```
could solve my question.

maybe errormsg is different between win7 and linux?

--- a/linter.py
+++ b/linter.py
@@ -4,7 +4,7 @@ from SublimeLinter.lint import Linter, util
 class Cppcheck(Linter):
     cmd = ('cppcheck', '--template=gcc', '--inline-suppr', '--quiet', '${args}', '${file}')
     regex = (
-        r'^(?P<file>.+\.[^:]+):(?P<line>\d+):((?P<col>\d+):)?\s+'
+        r'^(?P<file>(:\\|[^:])+):(?P<line>\d+):((?P<col>\d+):)?\s+'
         r'((?P<error>error)|(?P<warning>warning|style|performance|portability|information)):\s+'
         r'(?P<message>.+)'
     )

--- a/linter.py
+++ b/linter.py
@@ -4,7 +4,7 @@ from SublimeLinter.lint import Linter, util
 class Cppcheck(Linter):
     cmd = ('cppcheck', '--template=gcc', '--inline-suppr', '--quiet', '${args}', '${file}')
     regex = (
-        r'^(?P<file>.+):(?P<line>\d+):\s+'
+        r'^(?P<file>.+):(?P<line>\d+):(\d+):\s+'
         r'((?P<error>error)|(?P<warning>warning|style|performance|portability|information)):\s+'
         r'(?P<message>.+)'
     )

--- a/linter.py
+++ b/linter.py
@@ -4,7 +4,7 @@ from SublimeLinter.lint import Linter, util
 class Cppcheck(Linter):
     cmd = ('cppcheck', '--template=gcc', '--inline-suppr', '--quiet', '${args}', '${file}')
     regex = (
-        r'^(?P<file>.+):(?P<line>\d+):(\d+):\s+'
+        r'^(?P<file>.+\.[^:]+):(?P<line>\d+):((?P<col>\d+):)?\s+'
         r'((?P<error>error)|(?P<warning>warning|style|performance|portability|information)):\s+'
         r'(?P<message>.+)'
     )

